pkg/rancher-desktop/pages/containers/logs/_id.vue

```diff
@@ -0,0 +1,648 @@
+    BadgeState,
+    Banner,
+    LoadingIndicator,
+  },
+  data() {
+    return {
+      settings: undefined,
+      ddClient: null,
+      isLoading: true,
+      error: null,
+      containerName: '',
+      containerState: '',
+      isContainerRunning: false,
+      terminal: null,
+      fitAddon: null,
+      searchAddon: null,
+      streamProcess: null,
+      pendingLogs: '',
+      searchTerm: '',
+    };
+  },
+  computed: {
+    ...mapGetters('k8sManager', {isK8sReady: 'isReady'}),
+    containerId() {
+      return this.$route.params.id;
+    },
+    hasNamespaceSelected() {
+      return this.settings?.containerEngine?.name === ContainerEngine.CONTAINERD && this.settings?.containers?.namespace;
+    },
+  },
+  async mounted() {
+    this.$store.dispatch('page/setHeader', {
+      title: this.t('containers.logs.title'),
+      description: '',
+    });
+
+    ipcRenderer.on('settings-read', (event, settings) => {
+      this.settings = settings;
+      this.initializeLogs();
+    });
+
+    ipcRenderer.send('settings-read');
+    this.initializeLogs();
+
+    window.addEventListener('keydown', this.handleGlobalKeydown);
+  },
+  beforeDestroy() {
+    this.stopStreaming();
+    this.terminal?.dispose();
+    ipcRenderer.removeAllListeners('settings-read');
+    window.removeEventListener('keydown', this.handleGlobalKeydown);
+  },
+  methods: {
+    async initializeLogs() {
+      if (window.ddClient && this.isK8sReady && this.settings) {
+        this.ddClient = window.ddClient;
+        await this.getContainerInfo();
+        await this.fetchLogs();
+
+        if (this.isContainerRunning) {
+          await this.startStreaming();
+        }
+      }
+    },
+    async getContainerInfo() {
+      try {
+        const listOptions = {
+          all: true,
+          filters: `id=${this.containerId}`,
+        };
+
+        if (this.hasNamespaceSelected) {
+          listOptions.namespace = this.hasNamespaceSelected;
+        }
+
+        const containers = await this.ddClient?.docker.listContainers(listOptions);
+        const container = containers[0];
+
+        if (container) {
+          const name = Array.isArray(container.Names) ? container.Names[0] : container.Names.split(/\s+/)?.[0];
+          this.containerName = name?.replace(/_[a-z0-9-]{36}_[0-9]+/, '') || container.Id.substring(0, 12);
+          this.containerState = container.State || container.Status;
+          this.isContainerRunning = container.State === 'running' || container.Status === 'Up';
+        } else {
+          this.containerName = this.containerId.substring(0, 12);
+          this.containerState = 'unknown';
+          this.isContainerRunning = false;
+        }
+      } catch (error) {
+        console.error('Error getting container info:', error);
+        this.containerName = this.containerId.substring(0, 12);
+        this.containerState = 'unknown';
+        this.isContainerRunning = false;
+      }
+    },
+    async fetchLogs() {
```

Comment:
I don't think you need `fetchLogs` at all, since `startStreaming` can fetch the same logs (just drop the check if the container is running)?